core/src/main/java/com/google/cloud/sql/ConnectorConfig.java

```diff
@@ -108,7 +160,14 @@ public Builder withAdminServicePath(String adminServicePath) {
 
     /** Builds a new instance of {@code ConnectionConfig}. */
     public ConnectorConfig build() {
-      return new ConnectorConfig(targetPrincipal, delegates, adminRootUrl, adminServicePath);
+      return new ConnectorConfig(
```

Comment:
I think build should fail if multiple credentials have been set. WDYT?

Reply:
+1

Reply:
Good point. For named connectors, `ConnectorConfig.Builder.build()` will throw an IllegalStateException if more than one field is set. 

For unnamed connectors, it is only possible to set one of these fields, so there is no chance of creating a bad state.